File: 121. Best Time to Buy and Sell Stock/solution.go
func maxProfit(prices []int) int {
    if len(prices) == 0 {
        return 0
    }
    max := 0
    cur := prices[0]
    for i:=1; i<len(prices); i++ {
        if prices[i] > cur {
            if prices[i] - cur > max {
                max = prices[i] - cur
            }
        } else {
            cur = prices[i]
        }
    }
    return max
}

File: 121. Best Time to Buy and Sell Stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    if(prices.length < 2) return 0
    let curMin = prices[0]
    let profit = 0
    for(let i=1; i<prices.length; i++) {
        if(prices[i] - curMin > profit) {
            profit = prices[i] - curMin
        }
        if(prices[i] < curMin) {
            curMin = prices[i]
        }
    }
    return profit
};